### ISSUE TITLE
[critical] PROUD after update to 1.0.2

### ISSUE THREAD

**Issue by irritanterik** [athombv/homey#1107]
**What is your Homey version (Settings → System)?**
1.0.1

**What did you try to achieve?**
Update to 1.0.2
I started the update manually. After clicking the 'install' button a message appeared with a message like 'allready installing'. Then the update process started. Homey talked about its update, website went offline, purple ring of update development-mode (PROUD) started.

**What was the result?**
Homey is stuck on its purple ring. Can't recover from this situation.

After a few hours I tried several recover scenario's:
1, 10, 60, 600 seconds unpowerd. Restart upside-down, upside-down after 2, 4, 6, 8 seconds.
Without... success...

**Any other remarks**
about 10 other users reported simular issues.
My router log does not show Homey's mac requesting an ip. 
After every restart it starts with 10 seconds orange ring, then purple ring will follow.
If hold upside down somewhere within the first 10 seconds the ring goes to red, but then purple will follow within seconds. No countdown voice.

**Comment by MdeRhoter** [comment]
Exactly the same issue here

**Comment by irritanterik** [comment]
On slack other users have 'solved' their issue with this procedure: https://forum.athom.com/discussion/comment/18032/#Comment_18032
You will lose all flows, devices, settings etc though.

**Comment by MdeRhoter** [comment]
That'll be plan B - I don't fancy recreating 100+ flows, let alone re-pairing a house.

**Comment by irritanterik** [comment]
Update procedure now is confirmed succesfull. Users report the did not lost any data, flows, devices and are fully recovered.

**Comment by TedTolboom** [comment]
Same issue occured while updating my Homey.

Followed the following steps to recover (thanx to community support):
How to recover from PROUD (with MAC or UBUNTU):

1) Follow the steps mentioned in: https://forum.athom.com/discussion/comment/18032/#Comment_18032
2) Wait until you see the message: 
     Entered state 13
     Finished flashing: [MAC address]
3) Disconnect Homey from your MAC and reconnect to the power plug
4) Homey will start booting.. and after a while show the rainbow ring... at SW version 0.8.x
5) Now hold Homey upside down and wait at least 15 seconds (normally you should hear a countdown, in my case I didn't)...
6) Put Homey back on it's feed and follow the (new) recovery procedure.
7) You will get the option to keep user data or do a full reset. I used the keep user data...
8) Again follow the steps... an Homey will recover to v1.0.2 with keeping the devices / flows etc (at least in my case).

**Comment by nattlip** [comment]
well looks like the patient cured himself  . i had a recovery to today, because of all the other recoveries i am gratefull the procedure was clear today, thank all who helped today. 

https://forum.athom.com/discussion/2286/homey-v1-0-2#latest

**Comment by peaoo** [comment]
I also had the proud. Fixed it with this workaround.
I noticed that HomeyDash is the only app that lost its settings after Recovery.

**Comment by sossienl** [comment]
Also PROUD here, what a mess (again)

**Comment by bvdbos** [comment]
@sossienl : did you get the purple ring on stable firmware? Or did you choose to update to experimental firmware?

**Comment by robertraaijmakers** [comment]
Same here, also PROUD. Restore procedure didn't work last night. Will have to try again somewhere at the end of this week. All went well till step 5. But when connecting it to the power cable and turning it upside down the LED ring becomes solid red, but instead of resetting the ring becomes purple again while turned upside down.

**Comment by irritanterik** [comment]
My Homey is now recovered from the PROUD.

**Comment by WeeJeWel** [comment]
Hey guys,

We really need a Homey that's broken to investigate the issue. Could someone send their Homey to us so we can take a look at it? You'll get a brand new Homey back instantly.

Emile

**Comment by TedTolboom** [comment]
Hi Emile,

I will check with Will (WathLamers).
He has an PROUD Homey and asked me to recover it today...

I will forward your question to him.

Ted

**Comment by Willieku** [comment]
Good evening.
We wanted to recover the homey this evening, but at your request i will send it to you.
There is however no way to send it this evening whit postnl so it will be tomorrow. Or is there a other way to send it faster?   anyone?

**Comment by WeeJeWel** [comment]
@Willieku awesome! Where do you live? Maybe we can drive by if it's not too far from Enschede :)

**Comment by Willieku** [comment]
Waalwijk      dat is haast anderhalf uur rijden denk ik.
Ik heb wel koffie

Op 28 nov. 2016 19:19 schreef "Emile Nijssen" <notifications@github.com>:

**Comment by WeeJeWel** [comment]
Stefan passes through there at approx 17:00, would that work? If so, please tell him at stefan@athom.nl !

**Comment by Willieku** [comment]
Ja prima
Zet ik mijn adres hier op de mail?

Op 28 nov. 2016 19:27 schreef "Emile Nijssen" <notifications@github.com>:

**Comment by bvdbos** [comment]
No, don't :-) This is Github... Just send a message to stephan...

**Comment by bvdbos** [comment]
we're reading along :-)

**Comment by Willieku** [comment]
Got it
I have send a mail to stefan

**Comment by TedTolboom** [comment]
On request of @WeeJeWel a more detailed timeline of my purple colored issues:

- Using the Mac Homey desktop app
- checked for the SW v1.0.2 at Friday around 19:00
- started the update installation (without reboot on forehand, uptime around 4-5 days)
- Homey started downloading (blue ring)
- during the black screen (Homey is updating), if I recall correctly with purple ring, received 3 warnings: 2x 'Already downloading' and 1x checksum failure. (several reports where people got this error message, and were able to complete the update successfully)
- In all 3 cases, I could only close these pop-ups.
- afterwards no apparent progress; Homey kept it's purple ring; rotating and breathing (ademende) purple beam
- for a 3MB update, it took much longer than expected; feedback from other Slack users: 3-5 minutes max
- tried to refresh the Desktop app, but lost the connection with Homey at all
- after almost 2 hours of continuous purple ring, I pulled the plug (30 secs out), reconnected, brief orange ring (~ 5 seconds) and continued purple ring
- waited again for 30-60 minutes, tried another PTP
- after plugging in the 'power' cable, with 10 seconds waiting time followed by turning Homey upside down, it started being purple again
- another PTP, not waiting 10 seconds, and turning Homey upside down while it still is showing the orange ring, it will briefly show a red ring... but then starts being purple again....
- Left Homey for the night and morning (turned on, being a beacon of purple light)
- Followed JeroenBos22's successfull approach and recovered my Homey with the procedure I already mentioned above or the [original form post](https://forum.athom.com/discussion/comment/38144/#Comment_38144)

Additional note:
- During recovery, the recovery engine indicated that a new SW version was available. I did not see a message that this is an experimental update, nor did I select an option (in recovery) to install experimental updates...
- In [one of the forum posts](https://forum.athom.com/discussion/comment/38248/#Comment_38248) ChrisBoer indicates a similar issue with a brand new Homey, indicating that it is not only an "experimental" updating issue

**Comment by Willieku** [comment]
During update,   the black screen appeared.    Then,  after about 10 sec,
i lost contact.
I was not home,  i at another adres.
I could not make contact anymore.
When i came home; there was the purple ring

Op 28 nov. 2016 21:11 schreef "dT" <notifications@github.com>:

**Comment by Willieku** [comment]
@WeeJeWel   perhaps another clue.
since last weak i read about the homey desktop app , i use it more.
i can remind that i was wondering if the app would work if i was not home.
now i am not sure anymore, but i think i did used the desktop app tostrt the update

**Comment by bvdbos** [comment]
This one is solved? no reports of PROUD after 1.0.3 went stable afaik....

**Comment by Willieku** [comment]
i think its solved.
emile emailed they found a bug.  that was for over a week.

**Comment by TedTolboom** [comment]
@bvdbos At the risk of being "flamed" for below reaction.

No, I cannot state that this issue has been solved.

Nor do I appreciate the absolute lack of response / feedback of anyone of Athom on this issue. 
Yes, @WeeJeWel made a generic statement in the SW (that is software) update of week 48: "a small bugfix release that mostly fixes a bug where the updater would give errors (and in some cases a purple ring) when there are many apps loaded." 
But it is not proportional to the impact this issue had on many of us.

It did not re-occur with my update towards SW v1.0.3. Although I did see the feedback of many updates still showing error messages. 
Please keep in mind, it is a retail product now, available to the general public.... 
and yes it has an option to go for experimental updates...

**Comment by bvdbos** [comment]
@TedTolboom No flaming :-) 

But this happened on experimental and @WeeJeWel stated this was fixed for the next FW (which was 1.0.3). And of course I know how much impact this had on some of you ;-) And yes, there were different error-messages (though mostly minor) but no PROUD's anymore...

**Comment by WeeJeWel** [comment]
Hey guys,

The error messages were a symptom of the purple ring bug. So these
shouldn't happen anymore when you're updating from 1.0.3 or higher.

In the update TO v1.0.3 we've included a hotfix so that while the messages
still would appear, at least the Homey wouldn't get bricked.

Sorry for the inconvenience, but we only found this bug due to the number
of Homeys increasing and the amount of apps on a Homey increasing as well.
But it all should be fixed :)
